ocs_ci/framework/__init__.py

```diff
@@ -189,12 +189,13 @@ def _refresh_ctx(self):
         ]
         self.to_dict = self.cluster_ctx.to_dict
         if self.RUN.get("kubeconfig"):
-            logger.debug("switching kubeconfig")
```

Comment:
Is this switching context to the default cluster ?

Reply:
Real attribute refresh happens at line 187-189, the purpose of setting kubeconfig as part of env is for other entities in the framework which rely on environment variable. So , effectively this is switching the context to the cluster you wanted not necessarily the default context.